cranelift/codegen/src/machinst/reg.rs

```diff
@@ -48,10 +48,26 @@ pub fn first_user_vreg_index() -> usize {
 pub struct Reg(VReg);
 
 impl Reg {
+    /// Create a register from a physical register
+    // FIXME(const-hack) remove in favor of `From` impl
```

Comment:
Can you say more here -- I'm not that familiar with the current state of constification of everything; is there some feature this is waiting on to stabilize or...?

Reply:
well, the feature that would be waiting for is const associated functions in traits (therefore allowing `From::from` to be const) but that afaik is in "never type land". I can remove the note as it doesn't make much sense to hold our breath on it

Reply:
Yeah, I think that's best -- usually when we have a FIXME we want it to be actionable, or at least give sufficient context so that we know what we're waiting on (i.e.: if someone else walked up to this code tomorrow, would they know what they need in order to fix it?). It's fine to evaluate questions like this once new Rust features land in the future.

Reply:
the issue with constness is that theres no good way to detect constness workarounds after the fact (the whole reason for the const-hack comments) but usually you'd tie them to specific RFCs yeah

Reply:
Relevant RFC fresh off the press: https://github.com/rust-lang/rfcs/pull/3762